### PR TITLE
Use the testcontainers mattermost module

### PR DESCRIPTION
**Pull request by jespino**
This PR updates the contenirized e2e tests to use the testcontainers mattermost module. This module provides a more robust way to start and stop the mattermost server and database containers.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,6 @@
 module github.com/mattermost/mattermost-plugin-msteams
 
-go 1.21
-
-toolchain go1.21.6
```
Comment (lieut-data) [inline go.mod:5]: Can you elaborate on this change? Wondering if we want to follow the server pattern here which currently uses both directives?

File: server/testutils/containere2e/containere2e.go [hunk 2]
```diff
@@ -124,6 +124,7 @@ func NewE2ETestPlugin(t *testing.T, extraOptions ...mmcontainer.MattermostCustom
...
...0cnVlLCJtZmEiOnRydWUsImdvb2dsZV9vYXV0aCI6dHJ1ZSwib2ZmaWNlMzY1X29hdXRoIjp0cnVlLCJjb21wbGlhbmNlIjp0cnVlLCJjbHVzdGVyIjp0cnVlLCJtZXRyaWNzIjp0cnVlLCJtaHBucyI6dHJ1ZSwic2FtbCI6dHJ1ZSwiZWxhc3RpY19zZWFyY2giOnRydWUsImFubm91bmNlbWVudCI6dHJ1ZSwidGhlbWVfbWFuYWdlbWVudCI6dHJ1ZSwiZW1haWxfbm90aWZpY2F0aW9uX2NvbnRlbnRzIjp0cnVlLCJkYXRhX3JldGVudGlvbiI6dHJ1ZSwibWVzc2FnZV9leHBvcnQiOnRydWUsImN1c3RvbV9wZXJtaXNzaW9uc19zY2hlbWVzIjp0cnVlLCJjdXN0b21fdGVybXNfb2Zfc2VydmljZSI6dHJ1ZSwiZ3Vlc3RfYWNjb3VudHMiOnRydWUsImd1ZXN0X2FjY291bnRzX3Blcm1pc3Npb25zIjp0cnVlLCJpZF9sb2FkZWQiOnRydWUsImxvY2tfdGVhbW1hdGVfbmFtZV9kaXNwbGF5Ijp0cnVlLCJjbG91ZCI6ZmFsc2UsInNoYXJlZF9jaGFubmVscyI6dHJ1ZSwicmVtb3RlX2NsdXN0ZXJfc2VydmljZSI6dHJ1ZSwib3BlbmlkIjp0cnVlLCJlbnRlcnByaXNlX3BsdWdpbnMiOnRydWUsImFkdmFuY2VkX2xvZ2dpbmciOnRydWUsImZ1dHVyZV9mZWF0dXJlcyI6dHJ1ZX0sImlzX3RyaWFsIjpmYWxzZSwiaXNfZ292X3NrdSI6ZmFsc2V9IMay/e4rVqZ1yEluKxCtWQJK8iWdpADuWyETHJcCDMV8ouQK3n/ocJsg1y7INrbSPZDw6quohjblLN5MqHLQi0c+5yRYwzBhisJD6MFWxFCSg99eSXqIeudAfKVU+WOdZxWhyLzob14hOEfjvN/2hNSNyTV4hqhzk62L9vHzzZsgrFu+zYu4pA6Y4yzZF9FyVvHW241BkGq7ZecmyS6NQsq1jlAhkoBpdW9PCvFDfYS3+CwKtWNfebItc4e9JTbVpo75n++59WV2faQDfiMBf2bYwe6OxzJIZ258r8C2KMFD1uqpQohIoDS9ziygAu2voqgsQqm1Btf1hMtgFAOW7w=="),
```
Comment (lieut-data) [inline server/testutils/containere2e/containere2e.go:127]: nit: is the indentation off here?

**Review by lieut-data — CHANGES_REQUESTED** [review]
Just one request for clarification re: `go.mod` changes, otherwise looks good!

File: go.mod (hunk 1 above)
Comment (jespino) [reply go.mod:5]: Sorry, I didn't change it manually, it looks like Google Mod Tidy did those changes.

Comment (jespino) [reply go.mod:5]: Reverting the change, the linter complain if I don't run the go mod tidy, and that is what is changing the go mod for that.

**Review by lieut-data — APPROVED** [review]

**Comment by lieut-data** [comment]
We're slimming down the plugin to reach code complete for security review, so won't need this update.

**Pull request closed without merging**